pkg/hostap/monitor.go

```diff
@@ -140,17 +157,30 @@ func (m *InterfaceMonitor) handleHostapdEvent(eventStr string) error {
 	eventStrSlice := strings.Split(eventStr, " ")
 	if len(eventStrSlice) < 2 {
 		if isSolicitedEvent(eventStr) {
+			if strings.Contains(eventStr, statusReply) {
+				m.ifEAPState = getState(eventStr)
+				if err := m.updateInterfaceStatus(); err != nil {
+					level.Info(m.Logger).Log("op", "monitor", "error updating interface status", err)
+				}
+			}
 			return nil
 		}
 		level.Info(m.Logger).Log("hostapd-event", "unhandled event", m.IfName, eventStr)
 		return nil
 	}
+	defer func() {
+		if err := m.updateInterfaceStatus(); err != nil {
+			level.Info(m.Logger).Log("op", "monitor", "error updating interface status", err)
+		}
+	}()
 	eventKeyStr := eventStrSlice[0][strings.Index(eventStrSlice[0], ">")+1:]
 	switch eventKeyStr {
 	case staConnectedEvent, eapSuccessEvent:
 		return m.handleAuthenticateEvent(eventStrSlice[1])
-	case staDisconnectedEvent, eapFailureEvent:
+	case staDisconnectedEvent:
 		return m.handleDeAuthenticateEvent(eventStrSlice[1])
+	case eapFailureEvent:
+		m.logEvent(kapi.EventTypeWarning, "authentication failure for supplicant %s", eventStrSlice[1])
```

Comment:
I see we're logging a k8s Event for auth failure, interface up, interface down, but I don't see one for authentication or deauthentication.